### PR TITLE
Add Support for INA228

### PR DESCRIPTION
**Pull request by davids5**
Testing only!

Start with `ina228 -X -b 1 start` When connected to power brick 1 `ina228 -X -b 2 start`  When connected to power brick 2.

I am seeing higher the real current values.

A Load | A reported | Delta A
-- | -- | --
0.344 | 0.4563 | 0.1123
0.54 | 0.7306 | 0.1906
1.14 | 1.513 | 0.373
4.217 | 5.05 | 0.833

Measurements of the Pin 9-10 are 0.43 mA > than on the shunt.

### CODE REVIEW

File: src/drivers/drv_sensor.h [hunk 1]
```diff
@@ -142,26 +142,27 @@
...
-#define DRV_DIST_DEVTYPE_UAVCAN	0x89
+#define DRV_ACC_DEVTYPE_UAVCAN	0x81
+#define DRV_BARO_DEVTYPE_UAVCAN	0x82
+#define DRV_BAT_DEVTYPE_UAVCAN	0x83
+#define DRV_DIFF_PRESS_DEVTYPE_UAVCAN	0x84
+#define DRV_FLOW_DEVTYPE_UAVCAN	0x85
+#define DRV_GPS_DEVTYPE_UAVCAN	0x86
+#define DRV_GYR_DEVTYPE_UAVCAN	0x87
+#define DRV_IMU_DEVTYPE_UAVCAN	0x88
+#define DRV_MAG_DEVTYPE_UAVCAN	0x89
```
Comment (dagar) [inline src/drivers/drv_sensor.h:164]: Some of these (eg DRV_MAG_DEVTYPE_UAVCAN) might correspond to calibrations (device ids).

We should do a big cleanup of this list, maybe even split it by type.

File: src/drivers/power_monitor/ina228/ina228_params.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * INA228 Power Monitor Shunt
+ *
+ * @group Sensors
+ * @min 0.000000001
+ * @max 0.1
+ * @decimal 10
+ * @increment .000000001
+ */
+PARAM_DEFINE_FLOAT(INA228_SHUNT, 0.0005f);
```
Comment (dagar) [inline src/drivers/power_monitor/ina228/ina228_params.c:66]: Are people actually going to need to configure any of this or is it something that could be hardcoded as a board specific?

eg truly generic INA228 support vs Holybro v5x power module with board configuration

Comment (davids5) [reply src/drivers/power_monitor/ina228/ina228_params.c:66]: These values follow the PM not the board. By having them one driver can be used with any vendor parts chosen on their PM.

File: src/drivers/drv_sensor.h (hunk 1 above)
Comment (davids5) [reply src/drivers/drv_sensor.h:164]: I wondered if order preservation mattered and looked back at the last change you made and inferred it did not. If we want to classify them then we should have reserved blocks. But we can fix it.

File: src/drivers/power_monitor/ina228/ina228.cpp [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+		updateParams();
+	}
+
+	// read from the sensor
+	// Note: If the power module is connected backwards, then the values of _power, _current, and _shunt will be negative but otherwise valid.
+	bool success{true};
+	success = success && (read(INA228_REG_VSBUS, _bus_voltage) == PX4_OK);
+	// success = success && (read(INA228_REG_POWER, _power) == PX4_OK);
+	success = success && (read(INA228_REG_CURRENT, _current) == PX4_OK);
+	success = success && (read(INA228_REG_VSHUNT, _shunt) == PX4_OK);
```
Comment (davids5) [inline src/drivers/power_monitor/ina228/ina228.cpp:308]: Disable once working.

File: src/drivers/power_monitor/ina228/ina228.h [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+#define INA228_DEVICE_REV_ID_SHIFTS          (0)
+#define INA228_DEVICE_REV_ID_MASK            (0xf << INA228_DEVICE_REV_ID_SHIFTS)
+#define INA228_DEVICEREV_ID(v)               (((v) & INA228_DEVICE_REV_ID_MASK) >> INA228_DEVICE_REV_ID_SHIFTS)
+#define INA228_DEVICE_ID_SHIFTS              (4)
+#define INA228_DEVICE_ID_MASK                (0xfff << INA228_DEVICE_ID_SHIFTS)
+#define INA228_DEVICEID(v)                   (((v) & INA228_DEVICE_ID_MASK) >> INA228_DEVICE_ID_SHIFTS)
+
+
+
+#define INA228_SAMPLE_FREQUENCY_HZ           5
```
Comment (davids5) [inline src/drivers/power_monitor/ina228/ina228.h:288]: Return to 10

File: src/drivers/drv_sensor.h (hunk 1 above)
Comment (davids5) [reply src/drivers/drv_sensor.h]: Done

File: src/drivers/power_monitor/ina228/ina228.h (hunk 4 above)
Comment (davids5) [reply src/drivers/power_monitor/ina228/ina228.h]: Done

File: src/drivers/power_monitor/ina228/ina228.cpp (hunk 3 above)
Comment (davids5) [reply src/drivers/power_monitor/ina228/ina228.cpp]: Done

**Comment by davids5** [comment]
@dagar - Changes we discussed are in. In retrospect, this assume bus 1 and 2 for the general case (in  in rc.sensors) - which is fine as the current use cases. But might not be a general solution, as this is board wiring specific.

**Comment by davids5** [comment]
@dagar - back to board wiring specific

**Review by dagar — APPROVED** [review]

**Pull request merged**